scripts/experiment_1.py

```diff
@@ -0,0 +1,449 @@
+    ("Number of epochs required to orient the model",),
+]
+
+metric_titles = {
+    (
+        ppca_evaluation.OrientationMetrics.DISTANCE_TO_PERMUTATION_MATRIX,
+        ppca_evaluation.OrientationMetrics.MEAN_DISTANCE_TO_PERMUTATION_MATRIX,
+    ): "Distance to a Permutation Matrix",
+    (ppca_evaluation.OrientationMetrics.ORIENTED_DISTANCE_TO_A,): "Matrix Error",
+    # (ppca_evaluation.OrientationMetrics.ORIENTED_DISTANCE_TO_A_NORMALISED,),
+    (ppca_evaluation.OrientationMetrics.SIGMA_SQUARED_HAT,): r"$\hat{\sigma}^2$",
+    (ppca_evaluation.OrientationMetrics.DISTANCE_MU,): r"$\mu$ Error",
+    ("Difference between subspaces",): "Subspace Error",
+    # ("Normalised difference between subspaces",),
+    ("Number of epochs required to orient the model",): "Epochs Required for Orientation",
+    (
+        ppca_evaluation.OrientationMetrics.TRAIN_LOSS,
+        ppca_evaluation.OrientationMetrics.TEST_LOSS,
+    ): "Train and Test Loss",
+}
+
+
+@hy.config
+@dataclass
+class Experiment1Config:
+    """
+    The configuration of the synthetic data with subspace estimation experiment.
+
+    Members:
+        N: The number of points to generate.
+        n: the dimensionality of the observed space.
+        d: The dimensionality of the subspace
+        d_hat: The assumed number of components (the number of components fit by pPCA)
+        d_measured: The number of relevant factors we are interested in / have measurements of
+        noise_std: The scale of the gaussian noise to add to the observations
+        M: The number of samples to use for orientation of the mixing matrix
+        manifold_optimisation: Whether to orient the estimate constrained to the Stiefel manifold or not.
+        N_eval: Number of data points used for independent evaluation in the end
+        n_runs: How many times to fit each parameter setting to get a reliable estimate of the performance
+        loss: The loss to use for the orientation fitting.
+              Currently, only "mse" is supported.
+        learning_rate, batch_size, max_epochs, min_delta, patience, num_workers, gpus: Parameters for the Orientator
+        seed: The seed for the random generators
+    """
+
+    N: List[int] = field(default_factory=lambda: [1024])
+    n: List[int] = field(default_factory=lambda: [64])
+    d: List[int] = field(default_factory=lambda: [8])
+    d_hat: List[int] = field(default_factory=lambda: [8])
+    d_measured: List[int] = field(default_factory=lambda: [8])
+    noise_std: List[float] = field(default_factory=lambda: [0.1 ** (1 / 2)])
+    M: List[int] = field(default_factory=lambda: [1024])
+    manifold_optimisation: List[bool] = field(default_factory=lambda: [True])
+    N_eval: int = 1024
+    n_runs: int = 20
+    loss: str = "mse"
+    learning_rate: float = 1e-2
+    batch_size: int = 32
+    max_epochs: int = 512
+    min_delta: float = 1e-3
+    patience: int = 16
+    seed: int = 42
+    num_workers: int = 12
+    gpus: int = 1
+
+
+def fit(
```

Comment:
In a perfect world, this function would be ~5 different functions, but we can live with it for now.